src/Castle.Services.Logging.NLogIntegration/NLogLogger.cs

```diff
@@ -113,6 +122,84 @@ public virtual Core.Logging.ILogger CreateChildLogger(String name)
 			return Factory.Create(Logger.Name + "." + name);
 		}
 
+		/// <summary>
+		///   Logs a trace message.
+		/// </summary>
+		/// <param name="message"> The message to log </param>
+		public void Trace(string message)
+		{
+			Log(LogLevel.Trace, message);
+		}
+
+		/// <summary>
+		///   Logs a trace message.
+		/// </summary>
+		/// <param name="messageFactory"> Factory constructing lazily the message to log if the level is enabled </param>
+		public void Trace(Func<string> messageFactory)
+		{
+			if (IsTraceEnabled)
+			{
+				Log(LogLevel.Debug, messageFactory());
+			}
+		}
+
+		/// <summary>
+		///   Logs a trace message.
+		/// </summary>
+		/// <param name="exception"> The exception to log </param>
+		/// <param name="message"> The message to log </param>
+		public void Trace(string message, Exception exception)
+		{
+			Log(LogLevel.Trace, message, exception);
+		}
+
+		/// <summary>
+		///   Logs a trace message.
+		/// </summary>
+		/// <param name="format"> Format string for the message to log </param>
+		/// <param name="args"> Format arguments for the message to log </param>
+		public void TraceFormat(string format, params object[] args)
+		{
+			Log(LogLevel.Trace, format, args);
+		}
+
+		/// <summary>
+		///   Logs a trace message.
+		/// </summary>
+		/// <param name="exception"> The exception to log </param>
+		/// <param name="format"> Format string for the message to log </param>
+		/// <param name="args"> Format arguments for the message to log </param>
+		public void TraceFormat(Exception exception, string format, params object[] args)
+		{
+			if (IsTraceEnabled)
```

Comment:
Detail / question: You've removed the checks even for most of the `Trace`-specific methods... why not here, too?

Reply:
Fixing (same as above)